redbot/setup.py

```diff
@@ -128,16 +129,25 @@ def get_name() -> str:
         print(
             "Please enter a name for your instance,"
             " it will be used to run your bot from here on out.\n"
-            "This name is case-sensitive and can only include characters"
-            " A-z, numbers, underscores, and hyphens."
+            "This name is case-sensitive and should only include characters"
+            " A-z, numbers, underscores (_) and periods (.)."
         )
         name = input("> ")
-        if re.fullmatch(r"[a-zA-Z0-9_\-]*", name) is None:
+        if re.fullmatch(r"[A-Za-z0-9_\.\-]*", name) is None:
             print(
-                "ERROR: Instance name can only include"
-                " characters A-z, numbers, underscores, and hyphens!"
+                "ERROR: Instance names can only include characters A-z, numbers, "
+                "underscores (_) and periods(.)."
             )
-            name = ""
+        elif "-" in name and confirm(
+            "Hyphen (-) in instance names may cause issues. Are you sure you want to continue with this instance name?",
+            default=False,
+        ):
+            return name
+        else:
+            return name
+
+        print()  # new line for aesthetics
+        name = ""
```

Comment:
This needs the change so that Red still doesn't allow to set an empty name, let's just do it as we did before:
```suggestion
            name = ""
        elif "-" in name and not confirm(
            "Hyphen (-) in instance names may cause issues. Are you sure you want to continue with this instance name?",
            default=False,
        ):
            name = ""

        print()  # new line for aesthetics
```